### PR TITLE
build: upgrae protoc and grpc toolchain 

### PR DESCRIPTION
**Pull request by tisonkun**

### CODE REVIEW

File: golang/pkg/clusterpb/cluster.pb.go [hunk 1]
```diff
@@ -19,7 +19,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 // 	protoc-gen-go v1.28.1
-// 	protoc        v3.20.1
+// 	protoc        v4.25.1
```
Comment (ZuLiangWang) [inline golang/pkg/clusterpb/cluster.pb.go:22]: The version of `protoc` is `v4.25.1` rather than `v3.25.1`, is this as expected?

Comment (tisonkun) [reply golang/pkg/clusterpb/cluster.pb.go:22]: As long as Golang compiled, it should be fine.

I'm going to add some checks on this.

Comment (tisonkun) [reply golang/pkg/clusterpb/cluster.pb.go:22]: As you can see, the generated code is not changed

Comment (ZuLiangWang) [reply golang/pkg/clusterpb/cluster.pb.go:22]: No problem, but the version of `protoc` in `README.md` is `v3.25.1`,the generate code is `v4.25.1`, it is confusing, should we add some comment to explain this question?

Comment (tisonkun) [reply golang/pkg/clusterpb/cluster.pb.go:22]: Yes. We should use https://github.com/protocolbuffers/protobuf/releases/tag/v25.1. Let me update it.

Comment (tisonkun) [reply golang/pkg/clusterpb/cluster.pb.go:22]: Updated.

**Review by ZuLiangWang — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
Wait a bit for https://issues.apache.org/jira/browse/INFRA-25344 merged and we can test the staging process.

**Pull request merged**